services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HankkeenHakemuksetResponseDeserializer.kt

```diff
@@ -0,0 +1,84 @@
+package fi.hel.haitaton.hanke.hakemus
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.JsonDeserializer
+import com.fasterxml.jackson.databind.node.ObjectNode
+import fi.hel.haitaton.hanke.OBJECT_MAPPER
+import fi.hel.haitaton.hanke.allu.ApplicationStatus
+import java.time.ZonedDateTime
+
+class HankkeenHakemuksetResponseDeserializer : JsonDeserializer<HankkeenHakemuksetResponse>() {
```

Comment:
You don't have to create a custom deserializer for the whole response, just for the part Jackson can't deserialize without help. The class it's having trouble with is the Hakemusalue interface, but that might not be the best pick, since it's used elsewhere and it has no explicit type information. But you could take a step back and only implement a deserializer for the singular `HankkeenHakemusResponse`.

Reply:
You're right. I changed it as you suggested.